test/dates/io.jl

```diff
@@ -267,8 +291,7 @@ f = "y m d"
 @test_throws ArgumentError Dates.Date("1 1 32",f)
 @test_throws ArgumentError Dates.Date(" 1 1 32",f)
 @test_throws ArgumentError Dates.Date("# 1 1 32",f)
-# can't find 1st space delimiter,s o fails
-@test_throws ArgumentError Dates.Date("1",f)
+@test Dates.Date("1",f) == Dates.Date(1)
```

Comment:
Also, the original code was inconsistent with this rule as it trimmed whitespace automatically:

Before PR:
```
julia> Dates.Date("1 ", "y m d")  # Delimiter removed when removing whitespace
ERROR: ArgumentError: Delimiter mismatch. Couldn't find first delimiter, " ", in date string
 in parse(::String, ::Base.Dates.DateFormat) at ./dates/io.jl:169
 in Date(::String, ::String) at ./dates/io.jl:277
```